pkg/profile/artifact.go

```diff
@@ -94,6 +102,73 @@ func (pa *ProfilesArtifactsMaker) Make(installation profilesv1.ProfileInstallati
 	return pa.generateOutput(filepath.Join(profileRootdir, "profile-installation.yaml"), &installation)
 }
 
+func (pa *ProfilesArtifactsMaker) makeArtifacts(installation profilesv1.ProfileInstallation) ([]Artifact, error) {
+	path := installation.Spec.Source.Path
+	branchOrTag := installation.Spec.Source.Tag
+	if installation.Spec.Source.Tag == "" {
+		branchOrTag = installation.Spec.Source.Branch
+	}
+	def, err := getProfileDefinition(installation.Spec.Source.URL, branchOrTag, path, pa.GitClient)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get profile definition: %w", err)
+	}
+	var artifacts []Artifact
+
+	for _, artifact := range def.Spec.Artifacts {
+		if pa.nestedName != "" {
+			artifact.Name = filepath.Join(pa.nestedName, artifact.Name)
+		}
+
+		if err := validateArtifact(artifact); err != nil {
+			return nil, fmt.Errorf("validation failed for artifact %s: %w", artifact.Name, err)
+		}
+
+		t := -1
+		if artifact.Profile != nil {
+			profileRepoName := profileRepo(installation)
```

Comment:
I realised that in fact, we only have two builders. Which is Kustomize and Chart. In any other case, it's just a recursive call. So I don't have a builder for "Profiles".

This makes this part a bit more verbose though, so I'm thinking about what to do about that. :)